src/EditorFeatures/Test/Diagnostics/DiagnosticProviderTestUtilities.cs

```diff
@@ -49,7 +49,7 @@ private static IEnumerable<Diagnostic> GetDiagnostics(DiagnosticAnalyzer analyze
                     syntaxNodeAnalyzerService: nodeInBodyAnalyzerService,
                     hostDiagnosticUpdateSource: exceptionDiagnosticsSource,
                     cancellationToken: CancellationToken.None, 
-                    testOnly_DonotCatchAnalyzerExceptions: donotCatchAnalyzerExceptions);
+                    testOnly_RethrowAnalyzerException: donotCatchAnalyzerExceptions);
```

Comment:
Couple of things:

1. Rethrow exception flag is not set by tests verifying hardening against analyzer exceptions, those tests are already written to verify diagnostics for exceptions. If any of such test is still written to expect exceptions, it ought to be fixed as you suggested.

2. Rethrow exception is set to true by default for analyzer tests (and overridden to false by few hardening tests). We want any unexpected exception in product and test analyzers to be unhandled, and bring down the process with a diagnosable crash dump. Diagnostics for such exceptions are not too helpful, especially for intermittent failures. Even worse, if the test doesn't check the analyzer exception diagnostics, then it might silently ignore analyzer exceptions.